app/src/main/java/protect/card_locker/preferences/Settings.java

```diff
@@ -92,6 +92,14 @@ public String getColor() {
         return getString(R.string.setting_key_theme_color, mContext.getResources().getString(R.string.settings_key_system_theme));
     }
 
+    public int getPreferredColumnCount() {
+        var defaultSymbol = mContext.getResources().getString(R.string.settings_key_automatic_column_count);
+        var defaultColumnCount = mContext.getResources().getInteger(R.integer.main_view_card_columns);
+        var columnCountSetting = getString(R.string.setting_key_column_count, defaultSymbol);
+        // the pref may be unset or explicitly set to default
+        return columnCountSetting.equals(defaultSymbol) ? defaultColumnCount : Integer.parseInt(columnCountSetting);
```

Comment:
I think we should probably, if we can't parse the integer, fall back to the default. It *should* never happen, but Catima crashing at launch with no way to change the setting back, requiring a full data wipe would be horrible.

Reply:
agreed